tests/args/test_common.py

```diff
@@ -114,3 +113,49 @@ def _(value):
 
     [which] = collect_dumped_objects(result)
     assert which == "re"
+
+
+def test_same_name_for_step_arg_and_example_parameter(pytester):
+    """Test that using the same name for step arg and example parameter works as intended."""
+    pytester.makefile(
+        ".feature",
+        step_args_examples_params=textwrap.dedent(
+            """\
+            Feature: Test feature
+
+              Scenario Outline: Scenario 1
+                Given Action "A" is taken
+                Then Some other action "<action>" is taken
+
+                Examples: Actions
+                  | action |
+                  | B      |
+            """
+        ),
+    )
+    pytester.makepyfile(
+        textwrap.dedent(
+            """
+        from pytest_bdd import scenarios, given, then, parsers
+
+        scenarios('.')
+
+
+        @given(parsers.parse('Action "{action}" is taken'))
+        def take_action1(action):
+            print(f'take_action1:{action}')
```

Comment:
let's use `dumb_obj` and `collect_dumped_obj` instead, I made them for exactly this purpose